### PR TITLE
sriovnet: Add GetVfIndexByPciAddress API

### PR DESCRIPTION
**Pull request by moshe010**
Signed-off-by: Moshe Levi <moshele@mellanox.com>

### CODE REVIEW

File: sriovnet.go [hunk 1]
```diff
@@ -382,3 +383,21 @@ func FreeVfByNetdevName(handle *PfNetdevHandle, vfIndex int) error {
 func GetVfNetdevName(handle *PfNetdevHandle, vf *VfObj) string {
 	return vfNetdevNameFromParent(handle.PfNetdevName, vf.Index)
 }
+
```
Comment (paravmellanox) [inline sriovnet.go]: an small example is good to have here in the exported API, specially with the VF name such as 0000:06:00:01 or just 06:00:01 is good enough.

Comment (moshe010) [reply sriovnet.go]: Done

File: sriovnet.go [hunk 2]
```diff
@@ -382,3 +383,23 @@ func FreeVfByNetdevName(handle *PfNetdevHandle, vfIndex int) error {
...
+// GetVfIndexByPciAddress gets a VF PCI address (e.g '0000:03:00.4') and
+// returns the correlate VF index.
+func GetVfIndexByPciAddress(vfPciAddress string) (string, error) {
```
Comment (paravmellanox) [inline sriovnet.go]: @moshe010 In previously merged API GetVfPresentor() accepts vfIndex as int.
Here returned vIndex is string. Lets have it consistent to have index as 'int' or uint32.

File: sriovnet.go [hunk 3]
```diff
@@ -2,13 +2,14 @@ package sriovnet
...
 import (
 	"fmt"
-	"github.com/satori/go.uuid"
-	"github.com/vishvananda/netlink"
 	"log"
 	"net"
 	"path/filepath"
 	"strconv"
 	"strings"
+
```
Comment (paravmellanox) [inline sriovnet.go]: @moshe010 this change is not related.

go fmt with go1.12.4 doesnt complain about making above change.
Any reason to move them?

File: sriovnet.go [hunk 4]
```diff
@@ -382,3 +383,23 @@ func FreeVfByNetdevName(handle *PfNetdevHandle, vfIndex int) error {
...
+	if err != nil {
+		return "", err
+	}
+	for _, match := range matches {
+		tmp, err := os.Readlink(match)
+		if err != nil {
+			continue
+		}
+		if strings.Contains(tmp, vfPciAddress) {
+			return virtFnRe.FindStringSubmatch(match)[1], nil
```
Comment (paravmellanox) [inline sriovnet.go]: @moshe010 did I miss virtFnRe definition?
I didn't' notice in previous patch or this one.
